aesara/graph/fg.py

```diff
@@ -465,7 +465,7 @@ def change_node_input(
             self.outputs[i] = new_var
         else:
             r = node.inputs[i]
-            if check and not r.type.is_super(new_var.type):
+            if check and not r.type.in_same_class(new_var.type):
```

Comment:
Previously a method from the Type class was called here that returned true, but now a method from TensorType is called, which now returns false because it additionally checks for a shape match. So I changed the called method.